flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionServiceTest.java

```diff
@@ -177,25 +178,32 @@ public void testOldConfirmLeaderInformation() throws Exception {
 	public void testErrorForwarding() throws Exception {
 		new Context() {{
 			runTest(() -> {
-				final Exception testException = new Exception("test leader Exeption");
+				final Exception testException = new Exception("test leader exception");
 
-				testingLeaderElectionDriver.getFatalErrorHandler().onFatalError(testException);
+				testingLeaderElectionDriver.onFatalError(testException);
 
 				testingContender.waitForError(timeout);
 				assertThat(testingContender.getError(), is(notNullValue()));
-				assertThat(testingContender.getError().getMessage(), containsString(testException.getMessage()));
+				assertThat(testingContender.getError(), FlinkMatchers.containsCause(testException));
 			});
 		}};
 	}
 
 	@Test
-	public void testErrorHappenAfterStop() throws Exception {
+	public void testErrorIsIgnoredAfterBeingStop() throws Exception {
 		new Context() {{
 			runTest(() -> {
-				final Exception testException = new Exception("test leader Exeption");
+				final Exception testException = new Exception("test leader exception");
 
 				leaderElectionService.stop();
-				testingLeaderElectionDriver.getFatalErrorHandler().onFatalError(testException);
+				testingLeaderElectionDriver.onFatalError(testException);
+
+				try {
+					testingContender.waitForError(timeout);
```

Comment:
Is the timeout really 3 seconds long? This might be a bit long for this test case. I would suggest to wait at most 100 or even 50 ms.